holmes/plugins/toolsets/kubernetes_logs.py

```diff
@@ -0,0 +1,284 @@
+            docs_url="https://docs.robusta.dev/master/configuration/holmesgpt/toolsets/kubernetes.html#logs",
+            icon_url="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRPKA-U9m5BxYQDF1O7atMfj9EMMXEoGu4t0Q&s",
+            prerequisites=[CallablePrerequisite(callable=self.prerequisites_callable)],
+            tools=[
+                LoggingTool(self),
+            ],
+            tags=[ToolsetTag.CORE],
+        )
+        self._api_client = None
+        self._core_v1_api = None
+
+    def prerequisites_callable(self, config: Dict[str, Any]) -> Tuple[bool, str]:
+        """Verify Kubernetes client can connect to the cluster"""
+        try:
+            # TODO: support API KEY / BearerToken
+            # Try to load the kubeconfig and access the API
+            self._initialize_client()
+            self._api_client.call_api(
+                "/version",
+                "GET",
+                auth_settings=["BearerToken"],
+                response_type="object",
+                _return_http_data_only=True,
+            )
+            return True, ""
+        except Exception as e:
+            return False, f"Kubernetes client initialization error: {str(e)}"
+
+    def get_example_config(self):
+        return LoggingConfig().model_dump()
+
+    def _initialize_client(self):
+        if not self._api_client:
+            try:
+                try:
+                    config.load_incluster_config()
+                except config.ConfigException:
+                    config.load_kube_config()
+
+                self._api_client = client.ApiClient()
+                self._core_v1_api = client.CoreV1Api(self._api_client)
+            except Exception:
+                logging.error("Failed to initialize Kubernetes client", exc_info=True)
+                raise
+
+    def fetch_logs(self, params: FetchLogsParams) -> StructuredToolResult:
+        try:
+            self._initialize_client()
+
+            pod = self._find_pod(params.namespace, params.pod_name)
+            all_logs = []
+            if not pod or not pod.containers:
+                all_logs = self._fetch_pod_logs(
+                    params=params, previous=True
+                ) + self._fetch_pod_logs(params=params, previous=False)
+            else:
+                all_logs = self._fetch_pod_logs(
+                    params=params, containers=pod.containers, previous=True
+                )
+                all_logs = all_logs + self._fetch_pod_logs(
+                    params=params, containers=pod.containers, previous=False
+                )
+
+            if params.filter_pattern:
+                all_logs = self._filter_logs(all_logs, params.filter_pattern)
+
+            if params.limit and params.limit < len(all_logs):
+                all_logs = all_logs[-params.limit :]
+
+            # Format output
+            formatted_logs = self._format_logs(all_logs)
+
+            return StructuredToolResult(
```

Comment:
I think that in formatted logs we are return str even if no logs are found. In the structure tool response we also added support for no data.
We need to also support that case. e.g. no logs the status should not be success but no data.